scripts/heroku-build

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+# Build a Heroku app slug using the Heroku (classic) buildpack interface.
+# <https://devcenter.heroku.com/articles/buildpack-api>
+#
+set -euo pipefail
+
+cd "$(dirname "$0")/.."
+
+echo "---> Hello from $0"
+echo "     Source directory is $PWD"
+echo "     Build  directory is $PWD/build"
+echo
+echo "---> Removing any previous build/app/"
+rm -rf build/app/
+
+echo
+echo "---> Ensuring build dirs exist"
+mkdir -vp build/{app,cache,env}
+
+echo
+echo "---> Copying source to build/app/"
+# Don't use `git archive` because it excludes uncommitted (but unignored) files
+# in the working tree.  Pass thru tar so that it handles directory creation for
+# us.
+git ls-files --cached --others --exclude-standard -z \
+  | tar --create  --file - --null --files-from - \
+  | tar --extract --file - --verbose -C build/app/
+
+echo
+echo "---> Setting buildpack env vars from source"
+export SOURCE_VERSION SOURCE_DESCRIPTION
+echo "     SOURCE_VERSION=${SOURCE_VERSION:=$(git describe --always --dirty)}"
+echo "     SOURCE_DESCRIPTION=${SOURCE_DESCRIPTION:=$(git log -1 --format=%s)}"
+
+if [[ ! -d build/pack ]]; then
+  echo
+  echo "---> Cloning buildpack into build/pack/"
+  git clone --depth 1 https://github.com/heroku/heroku-buildpack-nodejs build/pack
+else
+  echo
+  echo "---> Using buildpack in build/pack/"
+fi
+
+echo
+echo "---> Buildpack version is $(git -C build/pack describe --always --dirty)"
+
+echo
+echo "---> Running buildpack"
+docker run \
+  --rm --interactive $(tty --quiet && echo --tty) \
+  --user $(id -u):$(id -g) \
+  --volume "$(realpath build/app)":/build/app:rw \
+  --volume "$(realpath build/pack)":/build/pack:ro \
+  --volume "$(realpath build/cache)":/build/cache:rw \
+  --volume "$(realpath build/env)":/build/env:ro \
+  --env STACK=heroku-20 \
+  --env SOURCE_VERSION \
+  --env SOURCE_DESCRIPTION \
+  --env NODE_MODULES_CACHE=false \
+  heroku/heroku:20-build bash -c '
```

Comment:
*note*

`heroku/heroku:20-build` (and `22-build`) are single-platform `linux/amd64` - no `linux/arm64` variant. This means `scripts/heroku-build` runs slower on Apple silicon, but not terribly slow - 7m17s on my M1 just now.

`heroku/heroku:24-build` is multi-platform with a `linux/arm64` variant. Just noting a reason to upgrade in the future.

Reply:
Good to note! Upgrading should be pretty painless, but I wanted to save that for another PR.